rest-api-spec/src/main/resources/rest-api-spec/test/search_pipeline/10_basic.yml

```
@@ -1,8 +1,8 @@
 ---
 "Test basic pipeline crud":
   - skip:
-      version: " - 2.99.99"
-      reason: "to be introduced in future release / TODO: change if/when we backport to 2.x"
+      version: " - 2.7.0"
```

```suggestion
      version: " - 2.6.99"
```

@msfroh sorry, probably yet another confusing question: if the feature is available in 2.7.0, we should skip all versions up to 2.7.0 (basically, using 2.6.99 placeholder as the 2.6 version range limit), does it make sense?

---

Looks like you are correct. 

I never understood the meaning of these version strings, so I had assumed that `-` was a silly way of saying `<`.  

I just walked through the `parse` method and constructor in the `SkipSection` class, which brought me to the `VersionRange` class. Now I understand that the `version` property is range or list of ranges, where each range is of the form `a - b`, meaning `[a, b]` (so it includes both endpoints), but where either `a` or `b` may be missing, which extends the given bound off to (negative) infinity. It makes sense now!